### PR TITLE
Add installation instructions for m4 on mac

### PR DESCRIPTION
**Pull request by stradig**
## Problem
Building on MacOS failed due to missing m4. Although a window was popping up claiming to install m4, this was not helping.

## Summary of changes
Add instructions to install m4 using brew and link it (thanks to Folke for helping).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +64,12 @@ brew install protobuf openssl flex bison icu4c pkg-config
 echo 'export PATH="$(brew --prefix openssl)/bin:$PATH"' >> ~/.zshrc
 ```
 
+If you get errors about missing `m4` you may have to install it manually:
+```
+brew install m4
+brew link --force m4
+````
```
Comment (cloneable) [inline README.md:71]: Needs one fewer tick.

**Review by cloneable — APPROVED** [review]

**Comment by stradig** [comment]
Thanks, VS Code sometimes adds this :(. Fixed.

**Comment by davidgomes** [comment]
> LGTM -- @davidgomes also added this in one of his PR, anyways, we can get this merged first

Just merge this, I will rebase and fix conflicts because my PR will take a while to get merged.

**Review by bayandin — CHANGES_REQUESTED** [review]
`m4` comes with command line tools (CLT, installed via `xcode-select --install`), @stradig / @davidgomes could you guys share your `brew config` output (it shows which CLT version is installed)

**Comment by davidgomes** [comment]
> `m4` comes with command line tools (CLT, installed via `xcode-select --install`), @stradig / @davidgomes could you guys share your `brew config` output (it shows which CLT version is installed)

It didn't for me, for sure. Anyway, here's my `brew config` output:

```
HOMEBREW_VERSION: 4.3.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1f9bd2de89883ec9a59db3eab0f5475bce06d5bc
Last commit: 3 days ago
Core tap JSON: 05 Sep 14:13 UTC
Core cask tap JSON: 05 Sep 14:13 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 11
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: 11-core 64-bit arm_lobos
Clang: 15.0.0 build 1500
Git: 2.45.2 => /opt/homebrew/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

**Comment by cloneable** [comment]
> `m4` comes with command line tools (CLT, installed via `xcode-select --install`), @stradig / @davidgomes could you guys share your `brew config` output (it shows which CLT version is installed)

No, it's a known issue for a few months now and Apple is working on a fix supposedly. `https://github.com/Homebrew/homebrew-core/issues/165388#issuecomment-2029794821`
I wonder if they initially did that because of M4 branding.

**Comment by stradig** [comment]
Thanks!

Here is my output

```
HOMEBREW_VERSION: 4.3.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1f9bd2de89883ec9a59db3eab0f5475bce06d5bc
Last commit: 3 days ago
Core tap JSON: 05 Sep 13:16 UTC
Core cask tap JSON: 05 Sep 13:16 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: dodeca-core 64-bit arm_lobos
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false

```

**Comment by bayandin** [comment]
> > `m4` comes with command line tools (CLT, installed via `xcode-select --install`), @stradig / @davidgomes could you guys share your `brew config` output (it shows which CLT version is installed)
> 
> No, it's a known issue for a few months now and Apple is working on a fix supposedly. `https://github.com/Homebrew/homebrew-core/issues/165388#issuecomment-2029794821` I wonder if they initially did that because of M4 branding.

Thanks for the link!
My knowledge was outdated 😅

**Pull request merged**